### PR TITLE
Projecting Mixed Reality Hands to Camera Views

### PR DESCRIPTION
**Pull request by sandrist**
- Created a mixed reality batch processing task for projecting 3D hand poses into the 2D image space of available HoloLens camera views, for visualization in PsiStudio: `ProjectHandsToCamerasTask`.
- Updates to `HandVisualizationObject` to no longer require adapters.
- Enabled the use of `AdjacentValueInterpolator{T}` inside Fuse/Join operators.
- Added methods for interpolating coordinate systems and hands.
- Fixed a bug in `Point3DGraphVisualizationObject` where node visuals were not being properly removed, because the logic for edge visuals was erroneously keeping them "alive".

### CODE REVIEW

File: Sources/MixedReality/Microsoft.Psi.MixedReality.Visualization.Windows/ProjectHandsToCamerasTask.cs [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+            // Get the tracked left and right hand streams
+            var hands = sessionImporter.OpenStream<(Hand Left, Hand Right)>(configuration.HandsStreamName);
+            var leftHand = hands.Select(h => h.Left).Where(h => h is not null && h.IsTracked);
+            var rightHand = hands.Select(h => h.Right).Where(h => h is not null && h.IsTracked);
+
+            foreach (var partition in sessionImporter.PartitionImporters.Values)
+            {
+                foreach (var streamMeta in partition.AvailableStreams)
+                {
+#pragma warning disable SA1101 // Prefix local calls with this
```
Comment (danbohus) [inline Sources/MixedReality/Microsoft.Psi.MixedReality.Visualization.Windows/ProjectHandsToCamerasTask.cs:41]: Is there a reason here for this pragma / not to stick with the stylecop rules?

File: Sources/MixedReality/Microsoft.Psi.MixedReality.Visualization.Windows/ProjectHandsToCamerasTask.cs [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+            get => this.handsStreamName;
+            set { this.Set(nameof(this.HandsStreamName), ref this.handsStreamName, value); }
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether to interpolate hands to the timestamps of camera messages.
+        /// </summary>
+        [DataMember]
+        [DisplayName("Interpolate Hands")]
+        [Description("Interpolate hands to the camera message times?")]
```
Comment (danbohus) [inline Sources/MixedReality/Microsoft.Psi.MixedReality.Visualization.Windows/ProjectHandsToCamerasTask.cs:144]: Indicates whether to ...

File: Sources/MixedReality/Microsoft.Psi.MixedReality/Operators.cs [hunk 3]
```diff
@@ -166,77 +166,87 @@ public static StereoKitColor32 ToStereoKitColor32(this SystemDrawingColor color)
...
+            var interpolatedJoints = new CoordinateSystem[numJoints];
+            for (int i = 0; i < numJoints; i++)
+            {
+                interpolatedJoints[i] = InterpolateCoordinateSystems(hand1.Joints[i], hand2.Joints[i], amount);
+            }
+
+            // If interpolating *any* non-zero amount of the pose of hand2 into the pose of hand1,
+            // then use values from hand2 for tracked, pinched, and gripped. Otherwise select from hand1.
+            return new Hand(
+                amount > 0 ? hand2.IsTracked : hand1.IsTracked,
```
Comment (danbohus) [inline Sources/MixedReality/Microsoft.Psi.MixedReality/Operators.cs:220]: Should this be 0.5?

File: Sources/MixedReality/Microsoft.Psi.MixedReality.Visualization.Windows/ProjectHandsToCamerasTask.cs (hunk 1 above)
Comment (sandrist) [reply Sources/MixedReality/Microsoft.Psi.MixedReality.Visualization.Windows/ProjectHandsToCamerasTask.cs:41]: It's a known bug in StyleCop: https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3012

File: Sources/MixedReality/Microsoft.Psi.MixedReality/Operators.cs (hunk 3 above)
Comment (sandrist) [reply Sources/MixedReality/Microsoft.Psi.MixedReality/Operators.cs:220]: I can switch to t > 0.5 for now

**Pull request merged**